app/services/change_schedule.rb

```diff
@@ -131,4 +132,10 @@ def validate_cannot_change_cohort
       errors.add(:cohort, I18n.t("cannot_change_cohort"))
     end
   end
+
+  def change_with_a_different_schedule
+    return unless new_schedule && participant_profile && new_schedule == participant_profile.schedule
+
+    errors.add(:schedule_identifier, I18n.t(:invalid_schedule))
```

Comment:
should we return an error message that indicates this schedule is already on the profile rather than invalid?